sklearn/utils/estimator_checks.py

```diff
@@ -573,12 +573,16 @@ def check_sample_weights_pandas_series(name, estimator_orig):
 def check_sample_weights_list(name, estimator_orig):
     # check that estimators will accept a 'sample_weight' parameter of
     # type list in the 'fit' function.
+    tags = _safe_tags(estimator_orig)
     if has_fit_parameter(estimator_orig, "sample_weight"):
         estimator = clone(estimator_orig)
         rnd = np.random.RandomState(0)
         X = pairwise_estimator_convert_X(rnd.uniform(size=(10, 3)),
                                          estimator_orig)
-        y = np.arange(10) % 3
+        if tags['binary_only']:
```

Comment:
unless `tags` is used somewhere else in the function, which does not appear to be the case, I think directly using `_safe_tags(estimator_orig)['binary_only']` is enough.

Or maybe just define `tags` right before using it, instead of at the top of the function.

Not very important though

Reply:
Used `_safe_tags(estimator_orig, 'binary_only')`